apps/api/src/admin/participant_manager.py

```diff
@@ -52,7 +52,15 @@ async def check_in_applicant(uid: str, associate: User) -> None:
     record: Optional[dict[str, object]] = await mongodb_handler.retrieve_one(
         Collection.USERS, {"_id": uid, "role": Role.APPLICANT}
     )
-    if not record or record["status"] not in (Status.ATTENDING, Status.CONFIRMED):
+    if not record:
+        raise ValueError
+
+    try:
+        status = record["status"]
+    except KeyError:
+        raise RuntimeError
+
```

Comment:
Rather than allowing a `KeyError` to happen, we can ensure the key exists with `"status" in record`, or use `dict.get` to retrieve a default value.
```suggestion
    status = record.get("status")
```